.github/workflows/development.yml

```diff
@@ -50,11 +65,25 @@ jobs:
       REACT_APP_MESSENGER_INVITE_PATH: ${{vars.MESSENGER_INVITE_PATH}}
       REACT_APP_ASSETS_PATH: ${{vars.ASSETS_PATH}}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
           node-version: '16'
       - run: echo '//npm.pkg.github.com/:_authToken=${{secrets.GITHUB_TOKEN}}' >> .npmrc
+
+      - name: Get node version
+        id: node
+        run: |
+          echo "version=$(node -v)" >> $GITHUB_OUTPUT
+
+      - name: Get node_modules cache
+        uses: actions/cache@v3.0.2
+        id: node_modules
+        with:
+          path: |
+            **/node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/package-lock.json') }}-${{ steps.node.outputs.version }}
```

Comment:
I wanted to use a composite action here (a common setup file), since this is repatead in each workflow

Something like:
```yml
name: Common Setup

on:
  workflow_run:
    workflows: ['Deploy Development', 'run-build', 'run-lint', 'run-tests', 'run-typecheck']
    types:
      - requested

jobs:
  common-setup:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v3

      - name: Setup Node.js
        uses: actions/setup-node@v3
        with:
          node-version: '16'

      - name: Get node version
        id: node
        run: echo "version=$(node -v)" >> $GITHUB_OUTPUT

      # rebuild the cache if OS, package-lock.json or node version changes
      - name: Get node_modules cache
        uses: actions/cache@v3.0.2
        id: node_modules
        with:
          path: |
            **/node_modules
          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/package-lock.json') }}-${{ steps.node.outputs.version }}
```

And then in the actual workflow use:
```yml
- uses: actions/checkout@v2
- uses: ./.github/workflows/common/setup
```

I tried a few things but that didn't work 🤔